source/common/stats/thread_local_store.cc

```diff
@@ -75,6 +91,29 @@ void ThreadLocalStoreImpl::shutdownThreading() {
   shutting_down_ = true;
 }
 
+void ThreadLocalStoreImpl::mergeHistograms(PostMergeCb mergeCb) {
+  if (!shutting_down_) {
+    tls_->runOnAllThreadsWithBarrier(
+        [this]() -> void {
+          for (ScopeImpl* scope : scopes_) {
+            for (auto histogram : tls_->getTyped<TlsCache>().scope_cache_[scope].histograms_) {
+              histogram.second->beginMerge();
+            }
+          }
+        },
+        [this]() -> void { mergeInternal(); });
+    mergeCb();
```

Comment:
You want to pass the callback here, not the invocation of the callback. In general at this point I think this is the last review cycle I'm going to do until there is full test coverage, because some of these issues would be really easy to spot with tests.